### PR TITLE
Adds HuggingFace pretrained encoder unit tests

### PR DESCRIPTION
**Pull request by geoffreyangus**

### CODE REVIEW

**Review by arnavgarg1 — APPROVED** [review]

**Review by jeffkinnison — APPROVED** [review]

**Review by justinxzhao — APPROVED** [review]

**Review by arnavgarg1 — APPROVED** [review]

**Review by jeffkinnison — APPROVED** [review]

File: tests/ludwig/encoders/test_text_encoders.py [hunk 1]
```diff
@@ -3,7 +3,61 @@
...
+    from huggingface_hub import HfApi
+
+    hf_api = HfApi()
+    hf_api.model_info(encoder_cls.DEFAULT_MODEL_NAME)
+
+
+@pytest.mark.parametrize("pretrained_model_name_or_path", ["bert-base-uncased"])
+@pytest.mark.parametrize("reduce_output", [None, "sum", "cls_pooled"])
+@pytest.mark.parametrize("max_sequence_length", [20])
+def test_auto_transformer_encoder(
```
Comment (jeffkinnison) [inline tests/ludwig/encoders/test_text_encoders.py:55]: Nice catch with the NoneType `AutoTransformerEncoder.DEFAULT_MODEL_NAME`. Could you add a short docstring explaining why this encoder has a separate test?

Comment (geoffreyangus) [reply tests/ludwig/encoders/test_text_encoders.py:55]: done!

Comment (jeffkinnison) [reply tests/ludwig/encoders/test_text_encoders.py:55]: Thanks!

**Review by jeffkinnison — APPROVED** [review]

**Review by tgaddair — APPROVED** [review]

**Review by jeffkinnison — APPROVED** [review]

**Comment by arnavgarg1** [comment]
Can you considering adding this one as well?

```
@pytest.mark.parametrize(
    "encoder_config_cls, encoder_cls",
    [
        (text_encoders_configs.ALBERTConfig, text_encoders.ALBERTEncoder),
        (text_encoders_configs.BERTConfig, text_encoders.BERTEncoder),
        (text_encoders_configs.XLMConfig, text_encoders.XLMEncoder),
        (text_encoders_configs.GPTConfig, text_encoders.GPTEncoder),
        (text_encoders_configs.RoBERTaConfig, text_encoders.RoBERTaEncoder),
        (text_encoders_configs.GPT2Config, text_encoders.GPT2Encoder),
        (text_encoders_configs.DistilBERTConfig, text_encoders.DistilBERTEncoder),
        (text_encoders_configs.TransformerXLConfig, text_encoders.TransformerXLEncoder),
        (text_encoders_configs.CTRLConfig, text_encoders.CTRLEncoder),
        (text_encoders_configs.CamemBERTConfig, text_encoders.CamemBERTEncoder),
        (text_encoders_configs.MT5Config, text_encoders.MT5Encoder),
        (text_encoders_configs.XLMRoBERTaConfig, text_encoders.XLMRoBERTaEncoder),
        (text_encoders_configs.LongformerConfig, text_encoders.LongformerEncoder),
        (text_encoders_configs.ELECTRAConfig, text_encoders.ELECTRAEncoder),
        (text_encoders_configs.FlauBERTConfig, text_encoders.FlauBERTEncoder),
        (text_encoders_configs.T5Config, text_encoders.T5Encoder),
        (text_encoders_configs.XLNetConfig, text_encoders.XLNetEncoder),
        (text_encoders_configs.DistilBERTConfig, text_encoders.DistilBERTEncoder),
    ],
)
def test_hf_default_reduce_output_strategy(
    encoder_config_cls: text_encoders_configs.BaseEncoderConfig, encoder_cls: text_encoders.HFTextEncoder
):
    """Test that default reduce_output strategy works for all text HFencoders."""
    reduce_output_strategy = encoder_config_cls.reduce_output
    encoder = encoder_cls(use_pretrained=False, reduce_output=reduce_output_strategy, max_sequence_length=20)
    inputs = torch.rand((2, 20)).type(encoder.input_dtype)
    encoder(inputs)
```

It may need some tweaking

**Review by arnavgarg1 — APPROVED** [review]

**Comment by arnavgarg1** [comment]
🚢

**Review by jeffkinnison — APPROVED** [review]

**Pull request merged**